docs/src/pages/docs/functions/helpers.md

```diff
@@ -280,8 +280,97 @@ Pass this function a function and it will return you a function that can be
 called in any form that you require until all arguments have been provided. For
 example if you pass a function: `f : (a, b, c) -> d` you get back a function
 that can be called in any combination, such as: `f(x, y, z)`, `f(x)(y)(z)`,
-`f(x, y)(z)`, or even `f(x)(y, z)`. This is great for doing partial application
-on functions for maximum re-usability.
+`f(x, y)(z)`, or even `f(x)(y, z)`. That is to say, this function fulfills the role
+of both curry and uncurry, returning a function that can be used as a curried
+function, an uncurried function, or any combination of argument applications in
+between. The returned function has `2^(n-1)` type signatures, where `n` is the
+number of parameters.
+
+```javascript
+import compose from 'crocks/helpers/compose'
+import curry from 'crocks/helpers/curry'
+import map from 'crocks/pointfree/map'
+import prop from 'crocks/maybe/prop'
+
+const crocksCurriedFunc =
+  curry((a, b, c, d) => a + b + c + d)
+
+crocksCurriedFunc(1)(2)(3)(4)  // [Fully curried]
+crocksCurriedFunc(1)(2)(3, 4)
+crocksCurriedFunc(1)(2, 3)(4)
+crocksCurriedFunc(1, 2)(3)(4)
+crocksCurriedFunc(1)(2, 3, 4)
+crocksCurriedFunc(1, 2, 3)(4)
+crocksCurriedFunc(1, 2, 3, 4)  // [Fully uncurried]
+crocksCurriedFunc(1, 2)(3, 4)
```

Comment:
These all need there output shown however given there structure and purpose maybe we can just put the output after the final line